pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -148,6 +152,12 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 	server.PodInfraContainerImage = imageTemplate.ExpandOrDie("pod")
 	server.CPUCFSQuota = true // enable cpu cfs quota enforcement by default
 
+	if dockerClient, _, err := docker.NewHelper().GetClient(); err == nil {
+		if docker.IsAPIVersionCompatible(dockerClient, minimumDockerAPIVersionForParallelPulls) {
```

Comment:
Another option seems like to call EnsureDocker in BuildKubernetesNodeConfig. I think having a handle to docker client seems a reasonable part of building node config. Also currently, the calls to BuildKubernetesNodeConfig and EnsureDocker in StartNode() are not very far apart (there is only one EnsureVolumeDir() in between, so not much).